File: MapsSDKExamplesObjc/Search/SearchCategoryViewController.m
/**
 * Copyright (c) 2018 TomTom N.V. All rights reserved.
 *
 * This software is the proprietary copyright of TomTom N.V. and its subsidiaries and may be used
 * for internal evaluation purposes or commercial use strictly subject to separate licensee
 * agreement between you and TomTom. If you are the licensee, you are only permitted to use
 * this Software in accordance with the terms of your license agreement. If you are not the
 * licensee then you are not authorised to use this software in any manner and should
 * immediately return it to TomTom N.V.
 */

#import "SearchCategoryViewController.h"
#import <TomTomOnlineSDKSearch/TomTomOnlineSDKSearch.h>

@interface SearchCategoryViewController() <TTSearchDelegate>
@property (nonatomic, strong) TTSearch *search;
@end

@implementation SearchCategoryViewController

- (BOOL)shouldDisplaySearchBar {
    return NO;
}

- (NSInteger)segmentsForControllSelected {
    return -1;
}

- (NSArray<NSString *> *)segmentsForControll {
    return @[@"PARKING", @"GAS", @"ATM"];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.search = [TTSearch new];
    self.search.delegate = self;
}

- (void)segmentChanged:(UISegmentedControl *)sender {
    if(!self.locationManager.lastLocation) {
        [self.toast toastWithMessage:@"Location not determined"];
        self.segmentedControl.selectedSegmentIndex = -1;
        [self.progress hide];
        return;
    }
    [self.progress show];
    switch (sender.selectedSegmentIndex) {
        case 2:
            [self searchCategoryATM];
            break;
        case 1:
            [self searchCategoryGas];
            break;
        default:
            [self searchCategoryParking];
            break;
    }
}

#pragma mark Examples

- (void)searchCategoryParking {
    TTSearchQuery *query = [[[[TTSearchQueryBuilder createWithTerm:@"parking"]
                              withCategory:YES]
                             withPosition:self.locationManager.lastLocation.coordinate]
                            build];
    [self.search searchWithQuery:query];
}

- (void)searchCategoryGas {
    TTSearchQuery *query = [[[[TTSearchQueryBuilder createWithTerm:@"gas"]
                              withCategory:YES]
                             withPosition:self.locationManager.lastLocation.coordinate]
                            build];
    [self.search searchWithQuery:query];
}

- (void)searchCategoryATM {
    TTSearchQuery *query = [[[[TTSearchQueryBuilder createWithTerm:@"atm"]
                              withCategory:YES]
                             withPosition:self.locationManager.lastLocation.coordinate]
                            build];
    [self.search searchWithQuery:query];
}

#pragma mark TTSearchDelegate

- (void)search:(TTSearch *)search completedWithResponse:(TTSearchResponse *)response {
    [self.progress hide];
    [self displayResults:response.results];
}

- (void)search:(TTSearch *)search failedWithError:(TTResponseError *)error {
    [self.toast toastWithMessage:[NSString stringWithFormat:@"error %@", error.userInfo[@"description"]]];
    [self.progress hide];
}

@end

File: MapsSDKExamplesSwift/Search/SearchCategoryViewController.swift
/**
 * Copyright (c) 2019 TomTom N.V. All rights reserved.
 *
 * This software is the proprietary copyright of TomTom N.V. and its subsidiaries and may be used
 * for internal evaluation purposes or commercial use strictly subject to separate licensee
 * agreement between you and TomTom. If you are the licensee, you are only permitted to use
 * this Software in accordance with the terms of your license agreement. If you are not the
 * licensee then you are not authorised to use this software in any manner and should
 * immediately return it to TomTom N.V.
 */

import UIKit
import MapsSDKExamplesCommon
import MapsSDKExamplesVC
import TomTomOnlineSDKSearch

class SearchCategoryViewController: SearchBaseViewController, TTSearchDelegate {
    
    let search = TTSearch()
    
    override func shouldDisplaySearchBar() -> Bool {
        return false
    }
    
    override func segmentsForControllSelected() -> Int {
        return -1
    }
    
    override func segmentsForControll() -> [String]  {
        return ["PARKING", "GAS", "ATM"]
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        search.delegate = self
    }
    
    override func segmentChanged(_ sender: UISegmentedControl) {
        guard locationManager.lastLocation != nil else {
            toast.toast(message: "Location not determined")
            segmentedControl!.selectedSegmentIndex = -1
            return
        }
        progress.show()
        switch sender.selectedSegmentIndex {
        case 2:
            searchCategoryATM()
        case 1:
            searchCategoryGas()
        default:
            searchCategoryParking()
        }
    }
    
    //MARK: Examples
    
    func searchCategoryParking() {
        let query = TTSearchQueryBuilder.create(withTerm: "parking")
            .withCategory(true)
            .withPosition(locationManager.lastLocation!.coordinate)
            .build()
        search.search(with: query)
    }
    
    func searchCategoryGas() {
        let query = TTSearchQueryBuilder.create(withTerm: "gas")
            .withCategory(true)
            .withPosition(locationManager.lastLocation!.coordinate)
            .build()
        search.search(with: query)
    }
    
    func searchCategoryATM() {
        let query = TTSearchQueryBuilder.create(withTerm: "atm")
            .withCategory(true)
            .withPosition(locationManager.lastLocation!.coordinate)
            .build()
        search.search(with: query)
    }
    
    //MARK: TTSearchDelegate
    
    func search(_ search: TTSearch, completedWith response: TTSearchResponse) {
        progress.hide()
        displayResults(response.results)
    }
    
    func search(_ search: TTSearch, failedWithError error: TTResponseError) {
        handleError(error)
    }
    
}
